### PR TITLE
Download smart photo functionality

### PR DESCRIPTION
**Pull request by snehaltembare**

### CODE REVIEW

File: app/src/main/java/com/snapxeats/DownloadTask.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+                    File downloadDirectory = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES)
+                            + "/" + mContext.getString(R.string.app_name) + "/" + mContext.getString(R.string.downloads) + "/");
+                    if (!downloadDirectory.exists()) {
+                        if (!downloadDirectory.mkdirs()) {
+                            return null;
+                        }
+                    }
+                    // Output stream to write file
+                    OutputStream output;
```
Comment (trivedidurgesh) [inline app/src/main/java/com/snapxeats/DownloadTask.java]: Separate the code for input and out put stream in 2 different methods

File: app/src/main/java/com/snapxeats/ui/restaurant/RestImagesAdapter.java [hunk 2]
```diff
@@ -44,16 +57,47 @@ public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
...
+        RestaurantPics restaurantPic = restaurantPicsList.get(position);
         Glide.with(mContext)
                 .load(restaurantPicsList.get(position).getDish_image_url())
                 .placeholder(R.drawable.ic_rest_info_placeholder)
                 .diskCacheStrategy(DiskCacheStrategy.NONE)
                 .override(Target.SIZE_ORIGINAL, Target.SIZE_ORIGINAL).thumbnail(THUMBNAIL)
                 .into(imageView);
 
+        //Check duplicate entry for dish to download
+        if (null != homeDbHelper) {
```
Comment (trivedidurgesh) [inline app/src/main/java/com/snapxeats/ui/restaurant/RestImagesAdapter.java]: move this hide unhide code in 4 different method one for each case

File: app/src/main/java/com/snapxeats/ui/restaurantInfo/RestaurantInfoActivity.java [hunk 3]
```diff
@@ -242,16 +284,57 @@ public void setUpRestaurantData() {
...
+
+                    mImgTextReview.setOnClickListener(this);
+                    mImgAudioReview.setOnClickListener(this);
+                    mImgDownload.setOnClickListener(this);
+
+                   mLayoutControls = itemView.findViewById(R.id.layout_controls);
+                    isImageTap = !isImageTap;
+                    if (isImageTap) {
+                        //Check duplicate entry for dish to download
+                        if (null != homeDbHelper) {
```
Comment (trivedidurgesh) [inline app/src/main/java/com/snapxeats/ui/restaurantInfo/RestaurantInfoActivity.java]: these if condition can be merged in single if try that.

File: app/src/main/java/com/snapxeats/ui/restaurantInfo/RestaurantInfoActivity.java [hunk 4]
```diff
@@ -242,16 +284,57 @@ public void setUpRestaurantData() {
...
+                });
         mRestInfoViewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
             @Override
             public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
-
+                if (null != mLayoutControls) {
+                    mLayoutControls.setVisibility(View.GONE);
+                }
+                //Check duplicate entry for dish to download
+                if (null != homeDbHelper) {
```
Comment (trivedidurgesh) [inline app/src/main/java/com/snapxeats/ui/restaurantInfo/RestaurantInfoActivity.java:323]: these if condition can be merged in single if try that

**Review by patilprajakta — APPROVED** [review]

**Pull request merged**